addons/point_of_sale/models/pos_config.py

```diff
@@ -555,6 +561,8 @@ def open_session_cb(self):
             self._check_company_payment()
             self._check_currencies()
             self._check_payment_method_receivable_accounts()
+            self._check_loyalties_product_availability()
```

Comment:
This should be moved to `pos_loyalty` module somehow. Your check should be done in `pos_loyalty`; `point_of_sale` doesn't need to be aware of them.

For example, in `pos_loyalty`:
```
    def open_session_cb(self):
        self.ensure_one()
        if not self.current_session_id:
            self._check_loyalties_product_availability()
        return super().open_session_cb()
```

Ideally, there should not be any change in `point_of_sale`

Reply:
Got it, but the "somehow" part is harder than I thought as no models in `pos_loyalty` inherit from the `pos_config`. Do you have any idea on the matter or should I forward it to your team and look at it at your own pace ?

Reply:
You can create a new class which inherits from the right model. For example like this: https://github.com/odoo/enterprise/blob/13.0/sale_intrastat/models/account_move.py